embassy-stm32/src/flash/mod.rs

```diff
@@ -0,0 +1,401 @@
+use crate::pac;
+use crate::peripherals::FLASH;
+use core::convert::TryInto;
+use core::marker::PhantomData;
+use core::ptr::write_volatile;
+use embassy::util::Unborrow;
+use embassy_hal_common::unborrow;
+
+use embedded_storage::nor_flash::{
+    ErrorType, NorFlash, NorFlashError, NorFlashErrorKind, ReadNorFlash,
+};
+
+const FLASH_BASE: usize = 0x8000000;
+
+#[cfg(flash_l4)]
+mod config {
+    use super::*;
+    pub(crate) const FLASH_SIZE: usize = 0x100000;
+    pub(crate) const FLASH_START: usize = FLASH_BASE;
+    pub(crate) const FLASH_END: usize = FLASH_START + FLASH_SIZE;
+    pub(crate) const PAGE_SIZE: usize = 2048;
+    pub(crate) const WORD_SIZE: usize = 8;
+}
+
+#[cfg(flash_wb)]
+mod config {
+    use super::*;
+    pub(crate) const FLASH_SIZE: usize = 0x100000;
+    pub(crate) const FLASH_START: usize = FLASH_BASE;
+    pub(crate) const FLASH_END: usize = FLASH_START + FLASH_SIZE;
+    pub(crate) const PAGE_SIZE: usize = 4096;
```

Comment:
perhaps the names `ERASE_SIZE`, `WRITE_SIZE` would be clearer. I've seen "page size" used as the write size in some places.